src/api/tojs.ts

```diff
@@ -1,15 +1,13 @@
-import {
-    isObservable,
-    isObservableArray,
-    isObservableMap,
-    isObservableObject,
-    isObservableValue,
-    keys
-} from "../internal"
+import { keys } from "./object-api"
+import { isObservable } from "./isobservable"
+import { isObservableArray } from "../types/observablearray"
+import { isObservableValue } from "../types/observablevalue"
+import { isObservableMap } from "../types/observablemap"
 
 export type ToJSOptions = {
     detectCycles?: boolean
     exportMapsAsObjects?: boolean
+    recurseEverything?: boolean
```

Comment:
I think it would be benificial to give a warning (or simply force) to set `detectCycles` when `recurseEverything` is used? Not sure

Reply:
I would prefer to force `detectCycles` to be true with presence of `recurseEverything` since it for sure will traverse deeply into all the levels, and it would be meaningful to stop when cycles are encountered. I will incorporate the changes in next commit. Thanks for idea!